tests/common.py

```diff
@@ -207,34 +215,42 @@ def assertNonObservedHolidayDates(self, *args):
 
     # Holiday name.
     def _assertHolidayName(self, name, instance_name, *args):
-        """Helper: assert a holiday with a specific name exists."""
+        """Helper: assert either a holiday with a specific name exists or
+        each holiday name matches an expected one depending on the args nature.
+        """
         holidays, items = self._parse_arguments(
             args,
             instance_name=instance_name,
         )
 
-        # Check the nature of the passed arguments.
-        arg = (
-            args[1]
-            if len(args) > 1 and issubclass(args[0].__class__, HolidayBase)
-            else args[0]
-        )
-        if isinstance(arg, (Generator, int, range, set)):
+        if len(items) == 0:  # A holiday name check.
+            self.assertFalse(holidays.get_named(name, lookup="exact"), name)
```

Comment:
```suggestion
            self.assertTrue(holidays.get_named(name, lookup="exact"), name)
```

Reply:
I've removed the dead code as it'd never be the case due to `raise_on_empty=True` in the `_parse_arguments` above.